src/Conversion/ONNXToKrnl/Math/Elementwise.cpp

```diff
@@ -1525,148 +1525,312 @@ static LogicalResult getPartiallyFlattenedSimdCode(
-  }
+  // To fuse Elementwise op with more one operands with the producer,
+  // the shape of the output the user Op has to have the same size
+  // output as that of the producer Op. Here dimension expansion with size
+  // 1 is allowed. Refer to hasNoBroadcast() definition.
+  // ToFix: This PR simply check static shape and does not use symbolic
+  // shape analysis and BroadcastShapeHelper
+
+  if (!hasStaticShape(defOp->getResults()[0].getType()))
+    return false;
 
-  bool checkFusedOp(Operation *op, SmallVector<Operation *, 2> &fusibleOps,
-      SmallVector<EmitScalarFunc, 2> &fuseEmitFunctions) {
-
-    // Notice: Though ClipOp is classified as unary element op in this file,
-    // ClipOp requires one required input and two optional input
-
-    return enqueueFusedOp<mlir::ONNXAbsOp, mlir::ONNXAtanOp, mlir::ONNXCastOp,
-        mlir::ONNXCeilOp, mlir::ONNXCosOp, mlir::ONNXCoshOp,
-        mlir::ONNXDequantizeLinearOp, mlir::ONNXEluOp, mlir::ONNXErfOp,
-        mlir::ONNXAcosOp, mlir::ONNXAcoshOp, mlir::ONNXAsinOp,
-        mlir::ONNXAsinhOp, mlir::ONNXAtanhOp, mlir::ONNXExpOp,
-        mlir::ONNXFloorOp, mlir::ONNXHardSigmoidOp, mlir::ONNXIsInfOp,
-        mlir::ONNXIsNaNOp, mlir::ONNXLeakyReluOp, mlir::ONNXLogOp,
-        mlir::ONNXNegOp, mlir::ONNXNotOp, mlir::ONNXReciprocalOp,
-        mlir::ONNXReluOp, mlir::ONNXRoundOp, mlir::ONNXSeluOp,
-        mlir::ONNXSigmoidOp, mlir::ONNXSignOp, mlir::ONNXSinOp,
-        mlir::ONNXSinhOp, mlir::ONNXSoftplusOp, mlir::ONNXSoftsignOp,
-        mlir::ONNXSqrtOp, mlir::ONNXTanOp, mlir::ONNXTanhOp>(
-        op, fusibleOps, fuseEmitFunctions);
+  auto defShape = getShape(defOp->getResults()[0].getType());
+  auto useShape = getShape(useOp->getResults()[0].getType());
+  if (defShape != useShape) {
+    return false;
   }
 
-  bool fusibleOpsIsEmpty() { return fusibleOps_.size() == 0; }
-
-  // This function starts for elementwise operation, op.
-  // A successor op (user) is fusible if it's the only user and an unary
-  // elementwise Op. The Op and its EmitScalarOpFor<T> are recorded into
-  // the vector.
-  void findFusibleOps() {
-    Operation *currentProducer = op_;
-    while (currentProducer->hasOneUse()) {
-      // Check the users is an unary elementwise op
-      // The right solution, I think, is to define EmitScalarOpFor as
-      // an interface of unary elementwise Ops.
-      // In the draft PR, variadic template is used to iterate through
-      // the possible ONNX Ops.
-      Operation *user = *currentProducer->getUsers().begin();
-      if (!checkFusedOp(user, fusibleOps_, fuseEmitFunctions_))
-        break;
+  for (size_t i = 0; i < useOp->getOperands().size(); i++) {
+    Operation *input = useOp->getOperand(i).getDefiningOp();
+    if (input == defOp)
+      continue;
 
-      currentProducer = user;
+    // Only input from block argument and constant is allowed.
+    if (!isa<BlockArgument>(useOp->getOperand(i)) &&
```

Comment:
`if (!isa<BlockArgument>(useOp->getOperand(i))` should be called before `Operation *input = useOp->getOperand(i).getDefiningOp();`. Otherwise, `input` can be null. For example,

```c
   Value inputVal = useOp->getOperand(i);

    // Only input from block argument and constant is allowed.
    if (!isa<BlockArgument>(inputVal)) {
        Operation *inputOp = inputVal.getDefiningOp();
        if (inputOp == defOp)
            continue;
        if(!isa<ONNXConstantOp>(input))
            return false;
    }
```

Reply:
Changed.
Yes, useOp->getOperand(i) could be null. But the test case works. Does it mean that null object is fine?